tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,494 @@
+    with patch('homeassistant.helpers.condition.dt_util.now',
+               return_value=now):
+        await async_media_play(hass, CLIENT_ENTITY_ID)
+        await hass.async_block_till_done()
+
+    state = hass.states.get(CLIENT_ENTITY_ID)
+    assert state.state == STATE_PLAYING
+
+    assert state.attributes.get(mp.ATTR_MEDIA_CONTENT_ID) == \
+        RECORDING['programId']
+    assert state.attributes.get(mp.ATTR_MEDIA_CONTENT_TYPE) == \
+        mp.MEDIA_TYPE_TVSHOW
+    assert state.attributes.get(mp.ATTR_MEDIA_DURATION) == \
+        RECORDING['duration']
+    assert state.attributes.get(mp.ATTR_MEDIA_POSITION) == 2
+    assert state.attributes.get(mp.ATTR_MEDIA_POSITION_UPDATED_AT) == now
+    assert state.attributes.get(mp.ATTR_MEDIA_TITLE) == RECORDING['title']
+    assert state.attributes.get(mp.ATTR_MEDIA_SERIES_TITLE) == \
+        RECORDING['episodeTitle']
+    assert state.attributes.get(mp.ATTR_MEDIA_CHANNEL) == \
+        "{} ({})".format(RECORDING['callsign'], RECORDING['major'])
+    assert state.attributes.get(mp.ATTR_INPUT_SOURCE) == RECORDING['major']
+    assert state.attributes.get(ATTR_MEDIA_CURRENTLY_RECORDING) == \
+        RECORDING['isRecording']
+    assert state.attributes.get(ATTR_MEDIA_RATING) == RECORDING['rating']
+    assert state.attributes.get(ATTR_MEDIA_RECORDED)
+    assert state.attributes.get(ATTR_MEDIA_START_TIME) == \
+        datetime(2018, 11, 10, 19, 0, tzinfo=dt_util.UTC)
+
+    # Test to make sure that ATTR_MEDIA_POSITION_UPDATED_AT is not
+    # updated if TV is paused.
+    with patch('homeassistant.helpers.condition.dt_util.now',
+               return_value=now + timedelta(minutes=5)):
+        await async_media_pause(hass, CLIENT_ENTITY_ID)
+        await hass.async_block_till_done()
+
+    state = hass.states.get(CLIENT_ENTITY_ID)
+    assert state.state == STATE_PAUSED
+    assert state.attributes.get(mp.ATTR_MEDIA_POSITION_UPDATED_AT) == now
+
+
+async def test_main_services(hass, platforms):
+    """Test the different services."""
+    main_media_entity, client_media_entity = await platforms(hass)
+
+    dtv_inst = main_media_entity.dtv
+
+    # DVR starts in turned off state.
+    state = hass.states.get(MAIN_ENTITY_ID)
+    assert state.state == STATE_OFF
+
+    # All these should call key_press in our class.
+    with patch.object(dtv_inst, 'key_press',
+                      wraps=dtv_inst.key_press) as mock_key_press, \
+        patch.object(dtv_inst, 'tune_channel',
+                     wraps=dtv_inst.tune_channel) as mock_tune_channel, \
+        patch.object(dtv_inst, 'get_tuned',
+                     wraps=dtv_inst.get_tuned) as mock_get_tuned, \
+        patch.object(dtv_inst, 'get_standby',
+                     wraps=dtv_inst.get_standby) as mock_get_standby:
+
+        # Turn main DVR on. When turning on DVR is playing.
+        await async_turn_on(hass, MAIN_ENTITY_ID)
+        await hass.async_block_till_done()
+        mock_key_press.assert_called_with('poweron')
```

Comment:
Don't use mock assert methods. A simple typo in the method name could make the assertion pass when it should fail. Import mock.call helper method and assert the `call_args` attribute on the mock instance.

Reply:
If we do a typo in the mock assert method name, it will be interpreted as a call to a mock mock method which will always pass. That can't happen in the latter case.

Reply:
Thus you mean if we do a typo within the test itself. For example:
`mock_key_press.assert_caled_with('poweron')`
Then the assert will pass no matter what. 

I just tested this.
`mock_key_press.assert_caled_with('poweron')`
fails with the error:
`AttributeError: assert_caled_with`

And if I try:
`mock_key_pres.assert_called_with('poweron')`
then it fails with `NameError: name 'mock_key_pres' is not defined`

And type in the patch.object itself (using 'key_pres' instead of 'key_press' gives me:
`AttributeError: <tests.components.media_player.test_directv.MockDirectvClass object at 0x106ddd898> does not have the attribute 'key_pres'`

And if I try:
`mock_key_press.asert_called_with('poweron')`
then I get: AttributeError: 'function' object has no attribute 'asert_called_with'

Not sure if something has changed. I did find:
https://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html

But I tried what they did there and it failed (assert_called_once in their example does work but it is also a valid assert from Python 3.6 onwards).